src/Test/Utilities/Portable/Metadata/MetadataReaderUtils.cs

```diff
@@ -428,6 +428,45 @@ internal static void VerifyMethodBodies(ImmutableArray<byte> peImage, Action<byt
             }
         }
 
+        internal static bool ContainsIsReadOnlyAttributes(this MetadataReader reader, CustomAttributeHandleCollection customAttributes, Accessibility accessibility)
+        {
+            return reader.ContainsAttribute(customAttributes, accessibility, "IsReadOnlyAttribute");
+        }
+
+        internal static bool ContainsIsByRefLikeAttributes(this MetadataReader reader, CustomAttributeHandleCollection customAttributes, Accessibility accessibility)
+        {
+            return reader.ContainsAttribute(customAttributes, accessibility, "IsByRefLikeAttribute");
+        }
+
+        private static bool ContainsAttribute(this MetadataReader reader, CustomAttributeHandleCollection customAttributes, Accessibility accessibility, string name)
```

Comment:
>MetadataReader [](start = 51, length = 14)

I think it would be better to use PEModule instance and rely on helpers within this type to find attributes based on specific AttributeDescription

Reply:
@AlekseyTs One downside of the helpers (`HasIsReadOnlyAttribute`, and the IsByRef one) is that they don't help check the accessibility. Are you ok with relaxing our tests that way?